docs/source/guides/models.mdx

```diff
@@ -1,217 +0,0 @@
-
-```python
->>> from optimum.neuron import NeuronModelForSequenceClassification
->>> model = NeuronModelForSequenceClassification.from_pretrained("my-neuron-repo")
-```
-
-As you see, there is no need to pass the neuron arguments used during the export as they are
-saved in a `config.json` file, and will be restored automatically by `NeuronModelForXXX` class.
-
-<Tip>
-
-When running inference for the first time, there is a warmup phase when you run the pipeline for the first time.  This run would take 3x-4x higher latency than a regular run.
-
-</Tip>
-
-## Discriminative NLP models
-
-As explained in the previous section, you will need only few modifications to your Transformers code to export and run NLP models:
-
-```diff
-from transformers import AutoTokenizer
--from transformers import AutoModelForSequenceClassification
-+from optimum.neuron import NeuronModelForSequenceClassification
-
-# PyTorch checkpoint
--model = AutoModelForSequenceClassification.from_pretrained("distilbert-base-uncased-finetuned-sst-2-english")
-
-# Compile your model during the first time
-+compiler_args = {"auto_cast": "matmul", "auto_cast_type": "bf16"}
-+input_shapes = {"batch_size": 1, "sequence_length": 64}
-+model = NeuronModelForSequenceClassification.from_pretrained(
-+    "distilbert-base-uncased-finetuned-sst-2-english", export=True, **compiler_args, **input_shapes,
-+)
-
-tokenizer = AutoTokenizer.from_pretrained("distilbert-base-uncased-finetuned-sst-2-english")
-inputs = tokenizer("Hamilton is considered to be the best musical of human history.", return_tensors="pt")
-
-logits = model(**inputs).logits
-print(model.config.id2label[logits.argmax().item()])
-# 'POSITIVE'
-```
-
-`compiler_args` are optional arguments for the compiler, these arguments usually control how the compiler makes tradeoff between the inference performance (latency and throughput) and the accuracy. Here we cast FP32 operations to BF16 using the Neuron matrix-multiplication engine.
-
-`input_shapes` are mandatory static shape information that you need to send to the neuron compiler. Wondering what shapes are mandatory for your model? Check it out
-with the following code:
-
-```python
->>> from transformers import AutoModelForSequenceClassification
->>> from optimum.exporters import TasksManager
-
->>> model = AutoModelForSequenceClassification.from_pretrained("distilbert-base-uncased-finetuned-sst-2-english")
-
-# Infer the task name if you don't know
->>> task = TasksManager.infer_task_from_model(model)  # 'text-classification'
-
->>> neuron_config_constructor = TasksManager.get_exporter_config_constructor(
-...     model=model, exporter="neuron", task='text-classification'
-... )
->>> print(neuron_config_constructor.func.get_mandatory_axes_for_task(task))
-# ('batch_size', 'sequence_length')
-```
-
-<Tip>
-
-Be careful, the input shapes used for compilation should be inferior than the size of inputs that you will feed into the model during the inference.
```

Comment:
I don't feel like this tip should be removed. We should remind users that padding are put in place, and compile with large static shapes means wasted compute.

Reply:
What you suggest is a bit different than what was written originally. Can you suggest a new tip and indicate where I should insert it ?

Reply:
something like:
```
Be careful, we pad the inputs to the shapes used for the compilation. But the inputs that you will feed into the model during the inference should have shapes inferior to the static shapes for compilation. And the padding brings computation overhead.
```

under the snippet of re-loading pre-compiled model I think (under line 247)